### PR TITLE
Tie detection added to the game.

### PR DESCRIPTION
**Pull request by Struckdown**
Adds tie detection to the game when neither player is able to win

### CODE REVIEW

File: src/main/java/cs/ualberta/cmput402/tictactoe/board/Board.java [hunk 1]
```diff
@@ -128,5 +133,7 @@ public Player getPlayerAtPos(int row, int col){
         return board[row][col];
     }
 
+    public int getMoveCount() {return moveCount;}
```
Comment (ZhimaoLin) [inline src/main/java/cs/ualberta/cmput402/tictactoe/board/Board.java]: You should make the coding style consistent with other functions such as getCurrentPlayer() and getWinner().

File: src/main/java/cs/ualberta/cmput402/tictactoe/board/Board.java [hunk 2]
```diff
@@ -11,6 +11,8 @@
     private Player currentPlayer;
     private Player winner;
     private Player board[][];
+    private int moveCount;
```
Comment (ZhimaoLin) [inline src/main/java/cs/ualberta/cmput402/tictactoe/board/Board.java:14]: I think it is better to reset moveCount in the initBoard() in order to make sure moveCount is reset to 0 every time a new game starts.

File: src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java [hunk 3]
```diff
@@ -34,6 +34,14 @@ public void playGame(){
 
         while (board.getWinner() == null){
             board.printBoard();
+
+            // If all moves have been played with no winner result game in tie
+
+            if(board.getMoveCount() == 9 ){
```
Comment (pshn111) [inline src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java]: You can add a more detail comment.
You also can delete line 39.

Comment (ZhimaoLin) [inline src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java]: Coding style is not consistent. 
There is no space after "if". There is extra space after 9.

**Review by ZhimaoLin — COMMENTED** [review]
Generally, the code looks good. 
Just some minor coding style problem.

**Review by ZhimaoLin — APPROVED** [review]
Looks good.

**Pull request merged**